### DISCUSSION TITLE
Подходит ли по концепции тут таблица циклов?

### DISCUSSION THREAD

**Discussion by simplepersonru** [totumonline/totum-ru-issues-and-discussions#368] [Как это сделать?] [answered]
Скрин из другой системы, аналог которой я делаю
![image](https://github.com/totumonline/totum-ru-issues-and-discussions/assets/71624171/0d94ea92-e70b-4c3e-adfb-5fb4400b6170)

Рассмотрим сущность "Арендатор". На картинке приведена карточка одного арендатора. Просматривать и изменять информацию по арендаторам хочется как из таблицы, так и из "карточки", плюс очень важна возможность фильтрации и сортировки. Вывод - в таблице Арендаторы и в карточке арендатора почти все поля повторяются 

я нашел 2 способа сделать это в тотум:

1. Через таблицу циклов. Соответственно Арендаторы - таблица циклов. Карточки арендатора - циклы. Но меня малость смутило в учебном курсе, что совсем не рассматривается вариант когда одни и те же поля есть и в таблице циклов и в самих циклах. Это нормальная практика? Внутрях PostgreSQL это inheritance таблиц (inheritance цикла от таблицы циклов) ?

2. Через временные таблицы. В таблице Арендаторы название арендатора будет с кодом действия по клику + c подсветкой аля url, а действие - linkToDataTable на временную таблицу арендатора. Минус этого подхода я увидел в том, что поскольку таблицы временные то сложно гарантировать что вкладка с этой таблицей будет актуальна например через сутки (у многих людей миллиард вкладок в браузере)

Подскажите каким из двух путей лучше идти и почему

**Comment by ttm-alexeypolunin** [comment, marked as answer]
Арендаторы — это таблица Циклов. Внутри карточка Арендатора — это таблица внутри цикла. Изнутри цикла часто данные достаются наверх в таблицу Циклов. Так-как это муторная операция для нее даже сахар сделали, который называется тип поля Ярлык.

В PG данные будут дублироваться — но Тотум глобально не экономит на спичках (точнее экономит, но не всегда). При изменении внитри цикла, строка в таблице циклов обновляется автоматически.

Со временной таблицей тоже можно, но тут явно тот случай, когда нужна таблица с Циклами. Временные делаются тогда, когда например сделали уже маленькое решение на основе простой таблицы, а потом через год эксплуатации поняли, что надо расширить. Вот тогда можно прикрутить через временные.

**Reply by erpic-ru** [reply]
Мне кажется, тут ещё есть вопрос: как редактировать данные и в таблице циклов, и в самом цикле.
Вот здесь есть ответ внизу страницы: https://ru.docs.totum.online/label-type-fields
